src/makeInfinityAdder.js

```diff
@@ -1,12 +1,23 @@
 'use strict';
 
-/***
- *
- * @return {function}
- */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(number) {
+    if (number) {
+      sum += number;
+
+      return adder;
+    } else {
+      const currentResult = sum;
+
+      sum = 0;
+
+      return currentResult;
+    }
+  };
```

Comment:
в этом случаи else не нужен
```suggestion
  function adder(number) {
    if (number) {
      sum += number;

      return adder;
    }
      const currentResult = sum;

      sum = 0;

      return currentResult;
  };
```